File: app/src/main/java/com/worldventures/dreamtrips/wallet/ui/settings/help/video/model/WalletVideoModel.java
package com.worldventures.dreamtrips.wallet.ui.settings.help.video.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.worldventures.core.modules.video.model.Video;
import com.worldventures.dreamtrips.wallet.ui.common.adapter.BaseViewModel;
import com.worldventures.dreamtrips.wallet.ui.settings.help.video.holder.VideoTypeFactory;

public class WalletVideoModel extends BaseViewModel<VideoTypeFactory> implements Parcelable {

   private final Video video;

   public WalletVideoModel(Video video) {
      this.video = video;
      this.modelId = video.getVideoName();
   }

   public Video getVideo() {
      return video;
   }

   @Override
   public int describeContents() { return 0; }

   @Override
   public void writeToParcel(Parcel dest, int flags) {
      dest.writeString(video.getImageUrl());
      dest.writeString(video.getVideoUrl());
      dest.writeString(video.getVideoName());
      dest.writeString(video.getCategory());
      dest.writeString(video.getDuration());
      dest.writeString(video.getLanguage());
   }

   protected WalletVideoModel(Parcel in) {
      final String imageUrl = in.readString();
      final String videoUrl = in.readString();
      final String name = in.readString();
      final String category = in.readString();
      final String duration = in.readString();
      final String language = in.readString();
      this.video = new Video(imageUrl, videoUrl, name, category, duration, language);
      this.modelId = name;
   }

   public static final Creator<WalletVideoModel> CREATOR = new Creator<WalletVideoModel>() {
      @Override
      public WalletVideoModel createFromParcel(Parcel source) {
         return new WalletVideoModel(source);
      }

      @Override
      public WalletVideoModel[] newArray(int size) {
         return new WalletVideoModel[size];
      }
   };

   @Override
   public int type(VideoTypeFactory typeFactory) {
      return typeFactory.type(this);
   }
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/video/model/WalletVideoModel.kt
package com.worldventures.wallet.ui.settings.help.video.model

import android.os.Parcel
import android.os.Parcelable
import com.worldventures.core.modules.video.model.Video
import com.worldventures.wallet.ui.common.adapter.BaseViewModel
import com.worldventures.wallet.ui.settings.help.video.holder.VideoTypeFactory

data class WalletVideoModel(val video: Video) : BaseViewModel<VideoTypeFactory>(), Parcelable {

   init {
      modelId = video.videoName
   }

   override fun describeContents() = 0

   override fun writeToParcel(dest: Parcel, flags: Int) {
      dest.writeString(video.imageUrl)
      dest.writeString(video.videoUrl)
      dest.writeString(video.videoName)
      dest.writeString(video.category)
      dest.writeString(video.duration)
      dest.writeString(video.language)
   }

   constructor(`in`: Parcel) : this(
         Video(`in`.readString(), `in`.readString(), `in`.readString(), `in`.readString(), `in`.readString(), `in`.readString())
   )

   override fun type(typeFactory: VideoTypeFactory): Int = typeFactory.type(this)

   companion object CREATOR : Parcelable.Creator<WalletVideoModel> {

      override fun createFromParcel(source: Parcel) = WalletVideoModel(source)

      override fun newArray(size: Int): Array<WalletVideoModel?> = arrayOfNulls(size)
   }
}
